src/commonMain/kotlin/fr/acinq/eclair/payment/IncomingPaymentHandler.kt

```diff
@@ -93,6 +93,54 @@ class IncomingPaymentHandler(val nodeParams: NodeParams, val walletParams: Walle
         return pr
     }
 
+    /**
+     * Save the "received-with" details of an incoming amount.
+     *
+     * - for a pay-to-open origin, we only save the id of the channel that was created for this payment.
+     * - for a swap-in origin, a new incoming payment must be created. We use the channel id to generate the payment's preimage.
+     * - for unknown origin, the amount is handled as a swap-in coming from an unknown address.
+     */
+    suspend fun process(channelId: ByteVector32, action: ChannelAction.Storage.StoreIncomingAmount) {
+        when (action.origin) {
+            null -> {
+                logger.warning { "incoming amount with empty origin, we store only minimal information" }
```

Comment:
It's unrelated to this PR, but there should be a comment explaining what the scenario is here. Why use a `SwapIn` origin with an empty addres?

Reply:
I use a `SwapIn` origin because in that case we are basically receiving an unsolicited amount. We could use a `KeySend` origin but since we don't really support keysend yet, it does not seems right.

Reply:
In the keysend scenario there is a preimage (it is provided by the sender), but here we are generating it. Shouldn't we remove that case altogether? We can deal with that later of course and just add this:
```suggestion
                // TODO: hacky, needs clean-up
                logger.warning { "incoming amount with empty origin, we store only minimal information" }
```

Reply:
So we would ignore this action and not store anything, right?

Reply:
I guess so but I haven't looked into it in detail. Let's deal with that in a separate PR? Doesn't seem very urgent?